connection.go

```diff
@@ -148,19 +146,27 @@ func (c *Connection) Close() {
 		c.Server.closes.Inc(1)
 	}
 
-	if c.Conn != nil {
-		// set guard deadline in case of bad connection
-		if err := c.Conn.SetDeadline(time.Now().Add(c.Server.Timeout)); err != nil {
-			_ = c.Conn.Close()
-		} else {
-			// flush write buffer before close
-			if c.Buffer.Writer != nil {
-				_ = c.Buffer.Writer.Flush()
-			}
-			_ = c.Conn.Close()
-		}
+	if err := c.Conn.SetDeadline(time.Now().Add(c.Server.Timeout)); err != nil {
```

Comment:
this was not done by me by the way =) I just moved the code. But is it done because there is a buffer flush after that, and without the deadline it could stuck